cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,150 @@
+
+    Option 2:
+
+    ```bash
+    rad resource-type new Mycompany.Messaging --schema-format yaml
+    ```
+    
+    Deb updates the schema definition with the required fields and properties for the messaging service Plaid. The authoring experience for the resource type will be covered in a separate document.
+
+2. Deb registers the resource type with Radius
+
+    Option 1 
+
+    ```bash
+    rad resource-provider create Mycompany.Messaging
+    ```
+    Pros : Follows the CRUDL convention and is more consistent with the other commands in Radius
+
+    Option 2
+
+    ```bash
+    rad resource-provider register Mycompany.Messaging
+    ```
+    Pros : Consistent with Recipe template registration to environment
+    Cons : Does not follow the CRUDL convention
+
+3. Deb creates the resource instance for the messaging service Plaid
+
+    ```bash
+    rad resource create Mycompany.Messaging/Plaid -f myplaid.json
+    ```
+    The resource `myplaid` of type `Mycompany.Messaging/Plaid`is created and available for use.
+
+4. Deb lists the resource providers and resource types available in Radius
+
+    ```bash
+    rad resource-provider list
+    ```
+    The list of resource providers in Radius are displayed.
+
+    ```bash
+    rad resource-type list --resource-provider Mycompany.Messaging
+    ```
+    The list of resource types under the resource provider `Mycompany.Messaging` are displayed.
+
+5. Deb makes changes to the resource type definition of `Plaid` and adds to the resource provider
+
+    Option 1:
+
+    ```bash
+    rad resource-provider create Mycompany.Messaging
+    ```
+    Pros : Same command can be used to update the resource provider with the new or updated resource type definitions
+
+    Option 2 :
+
+    ```bash
+    rad resource-type create -f plaid.yaml 
+    ```
+    Pro : No need to follow a rp directory structure. Resource type gets created/updated to the resource provider. 
+    Con : Resource provider is a required input in the `plaid.yaml` schema definition
+
+    Comment: It may be good and complete to support both these options.
+
+6. Deb wants to know the details of resource provider
+
+    ```bash
+    rad resource-provider show Mycompany.Messaging
+    ```
+    The details of the resource provider `Mycompany.Messaging` are displayed.
+
+7. Deb deletes a resource-type Plaid from the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-type delete Mycompany.Messaging/Plaid
+    ```
+    The resource type `Plaid` is deleted from the resource provider `Mycompany.Messaging`
+
+8. Deb deletes the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-provider delete Mycompany.Messaging
+    ```
+
+
+## Finalized set of commands 
+
+rad resource-provider 
+
+rad resource-type
+
+rad resource create
```

Comment:
Should we separate `rad resource create` from this doc? It's not really a UDT feature. It's just functionality we're missing that we happen to **need** now.